### PR TITLE
Updated Netsparker Parser

### PR DESCRIPTION
**Pull request by kiliczsh**
Fields added:
- Request
- Response
- CVSSv3

### CODE REVIEW

**Review by damiencarol — CHANGES_REQUESTED** [review]
Looks good to me.
The only missing things here is new checks in the unit tests to validate the new data

**Comment by kiliczsh** [comment]
Hi @damiencarol,

I have added 3 sample scan data and created unit test for Netsparker Parser.

File: dojo/unittests/tools/test_netsparker_parser.py [hunk 1]
```diff
@@ -13,36 +13,36 @@ def test_parse_file_with_one_finding(self):
         with self.subTest(i=0):
             finding = findings[0]
             self.assertEqual("Medium", finding.severity)
-            self.assertEqual(16, finding.cwe)
+            self.assertEqual('16', finding.cwe)
```
Comment (damiencarol) [inline dojo/unittests/tools/test_netsparker_parser.py:16]: today, `cwe` attribute is a number.

File: dojo/unittests/tools/test_netsparker_parser.py [hunk 2]
```diff
@@ -13,36 +13,36 @@ def test_parse_file_with_one_finding(self):
...
         findings = parser.get_findings(testfile, Test())
         self.assertEqual(40, len(findings))
         with self.subTest(i=0):
             finding = findings[0]
             self.assertEqual("Medium", finding.severity)
-            self.assertEqual(16, finding.cwe)
+            self.assertEqual('16', finding.cwe)
             self.assertIsNotNone(finding.description)
             self.assertGreater(len(finding.description), 0)
             self.assertEqual("5.5", finding.cvssv3)
```
Comment (damiencarol) [inline dojo/unittests/tools/test_netsparker_parser.py:32]: the CVSS v3 vector is still not ok

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/unittests/tools/test_netsparker_parser.py (hunk 1 above)
Comment (kiliczsh) [reply dojo/unittests/tools/test_netsparker_parser.py:16]: @damiencarol I checked it from [this action run](https://github.com/DefectDojo/django-DefectDojo/runs/2912823597). It was failed as this value was a string.

```
Traceback (most recent call last):
2021-06-25T09:49:04.7111523Z [36muwsgi_1         |[0m   File "/app/dojo/unittests/tools/test_netsparker_parser.py", line 16, in test_parse_file_with_one_finding
2021-06-25T09:49:04.7112747Z [36muwsgi_1         |[0m     self.assertEqual(16, finding.cwe)
2021-06-25T09:49:04.7113639Z [36muwsgi_1         |[0m AssertionError: 16 != '16'
```

File: dojo/unittests/tools/test_netsparker_parser.py (hunk 2 above)
Comment (kiliczsh) [reply dojo/unittests/tools/test_netsparker_parser.py]: Fixed this now 👍🏼

**Comment by damiencarol** [comment]
@kiliczsh your PR is good but you must switch `cwe` attribute to `int` or it will fail on save

**Comment by kiliczsh** [comment]
> @kiliczsh your PR is good but you must switch `cwe` attribute to `int` or it will fail on save

@damiencarol CWE was implemented before so only unit tests was failing but saving process was successful. I changed test for integer case. Previous tests was successful. I have setup for this patch and imports were successful.

**Comment by damiencarol** [comment]
@kiliczsh I really think that the parser need to convert CWEs to int. I understand it was the old behavior of the parser but could you fix this issue? I think we can accept and merge it after.

**Comment by kiblik** [comment]
Hi. Using `url` in finding is [deprecated](https://github.com/DefectDojo/django-DefectDojo/blob/44ab2d979f5d97b330258b7fd0f30f18d9553e5a/dojo/models.py#L1590-L1594). We are using endpoint now.

I suppose:
- it should be removed from [here](https://github.com/DefectDojo/django-DefectDojo/pull/4711/files#diff-2eb8d9651a662bc2c796080d1b3e12a6c8b2891a33b2f3081417ae2343f54b91R61)
- also [here](https://github.com/DefectDojo/django-DefectDojo/pull/4711/files#diff-2eb8d9651a662bc2c796080d1b3e12a6c8b2891a33b2f3081417ae2343f54b91R72)
- add as endpoint, like [here](https://github.com/DefectDojo/django-DefectDojo/blob/44ab2d979f5d97b330258b7fd0f30f18d9553e5a/dojo/tools/generic/parser.py#L49-L59)
- cleaned like [here](https://github.com/DefectDojo/cookiecutter-scanner-parser/blob/fa13c1c23b82599b99a00d7c96a27fd2df5bf436/temp-%7B%7B%20cookiecutter.tool_directory_name%20%7D%7D/unittests/tools/test_%7B%7B%20cookiecutter.tool_directory_name%20%7D%7D_parser.py#L20-L22)
- and also some unittest

**Comment by kiliczsh** [comment]
@damiencarol Changed CWE value from string to integer. 
@kiblik Created issue for updating url to endpoints. #4719

**Review by damiencarol — APPROVED** [review]
It's a good patch that improve quality and implement more features of findings. 👍

File: dojo/tools/netsparker/parser.py [hunk 3]
```diff
@@ -57,23 +59,29 @@ def get_findings(self, filename, test):
...
+            response = item["HttpResponse"]["Content"]
+
+            finding = Finding(title=title,
+                              test=test,
+                              description=findingdetail,
+                              severity=sev.title(),
+                              mitigation=mitigation,
+                              impact=impact,
+                              references=references,
+                              url=url,
```
Comment (damiencarol) [inline dojo/tools/netsparker/parser.py:72]:
```suggestion
```

**Comment by damiencarol** [comment]
@kiblik removed `url` as you suggested. I think it was a copy/paste more than a modification to add it.

**Comment by kiliczsh** [comment]
@damiencarol I think this failure was caused by another development?

```
2021-06-26T13:26:01.5934979Z [36muwsgi_1         |[0m ======================================================================
2021-06-26T13:26:01.5936081Z [36muwsgi_1         |[0m FAIL: test_is_defined (dojo.unittests.test_apiv2_methods.ApiEndpointMethods)
2021-06-26T13:26:01.5937309Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2021-06-26T13:26:01.5938180Z [36muwsgi_1         |[0m Traceback (most recent call last):
2021-06-26T13:26:01.5939176Z [36muwsgi_1         |[0m   File "/app/dojo/unittests/test_apiv2_methods.py", line 23, in test_is_defined
2021-06-26T13:26:01.5940390Z [36muwsgi_1         |[0m     self.assertIsNotNone(self.schema["paths"][BASE_API_URL + '/' + reg + '/'].get(method),
2021-06-26T13:26:01.5941596Z [36muwsgi_1         |[0m AssertionError: unexpectedly None : Endpoint: roles, Method: post
2021-06-26T13:26:01.5942410Z [36muwsgi_1         |[0m 
2021-06-26T13:26:01.5943196Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2021-06-26T13:26:01.5943989Z [36muwsgi_1         |[0m Ran 1467 tests in 1067.045s
2021-06-26T13:26:01.5944563Z [36muwsgi_1         |[0m 
2021-06-26T13:26:01.5945234Z [36muwsgi_1         |[0m FAILED (failures=1, skipped=168)
```

**Comment by kiblik** [comment]
Looks like site-effect of https://github.com/DefectDojo/django-DefectDojo/pull/4637

it was waiting too long :) I will check it

**Comment by kiblik** [comment]
Unittest fixed in #4721

**Comment by damiencarol** [comment]
I guess we need to rebase this PR ? @kiblik @valentijnscholten

**Comment by kiblik** [comment]
@damiencarol - yes, merge or rebase

Give me a moment, I'm going to fix also endpoints there (I will prepare PR in to this branch)

**Comment by kiblik** [comment]
https://github.com/kiliczsh/django-DefectDojo/pull/1

**Review by damiencarol — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**